### ISSUE TITLE
Refactor the code to make sure os path related logic is OS agnostic

### ISSUE THREAD

**Issue by arkid15r** [OWASP/Nettacker#933]
Refactor the code to make the code expecting Unix/Linux paths OS-independent.
Test and ensure that the the entire application with all its functionality (including Scan engine, API and WebUI) works equally fine on both Windows and Unix-like operating systems

**Comment by winters0x64** [comment]
Hi, I'd like to work on this issue, could you assign it to me?

**Comment by bhagyashree980** [comment]
Hi @arkid15r ,

I noticed that issue #933: Refactor the code to make sure os path related logic is OS agnostic #933 is currently assigned but hasn’t had any recent activity in the last two weeks. I’m eager to contribute and would love to help move this forward.
If the current assignee is unavailable or needs assistance, I would be happy to take over or collaborate to resolve this issue promptly. Please let me know how I can assist.
Looking forward to your response!

**Comment by winters0x64** [comment]
I'm still working on this, just wanted to make sure everything is working before i send the final PR.

**Comment by bhagyashree980** [comment]
@winters0x64 Thank you for your update

**Comment by winters0x64** [comment]
@bhagyashree980 , appreciate you looking into this, I was occupied with implementing some security checks, that's why this got delayed. If you want to contribute you can try to improve the nettacker engine part, I'll  be looking at that after this maybe we can collaborate on that.

**Comment by bhagyashree980** [comment]
Hi @winters0x64 ,
Thank you for the update and for sharing the context. I understand how implementing security checks can take priority. I appreciate the suggestion to look into improving the Nettacker engine part—I'll start exploring it right away.

**Comment by vishakha-mali** [comment]
It's been a month for the previous user,Can I get the chance?

**Comment by winters0x64** [comment]
@vishakha-mali , I appreciate your interest in the issue but I'm still resolving this issue with @arkid15r , my PR is almost complete, just testing all the edge cases that might arise due to cross platform compatibility issues.

**Comment by vishakha-mali** [comment]
Yeah sure, I thought you were not active.I just wanted to give a try as I am a beginner.

**Comment by winters0x64** [comment]
@vishakha-mali , I encourage you to still try to fix this issue, collective efforts make the project better, you'll also learn a lot. Feel free to ask any doubts if you have I'd be happy to help, all the best :)

**Comment by winters0x64** [comment]
@pUrGe12 I was not planning to migrate the Dockerfile to support windows as well, only going for the paths right now,  afaik it'll work on windows machine right now but if it affects upgradeability, as in future the path could become an issue so its better to use the proper paths for the respective OS. Sure we can work on this togethter, hmu.

**Comment by SumraShahzadi** [comment]
As a beginner in open source contribution ,I want to try to resolve this issue and I am working on it

**Comment by Manushya-a** [comment]
hey @SumraShahzadi I wanted to ask if you were working on this issue or if i could jump on this one?

**Comment by SumraShahzadi** [comment]
yes ,you can

**Comment by zekeyeager-tech** [comment]
Hi, I noticed this issue is unassigned, and I would love to contribute to resolving it.  
Can I take this up? Please let me know how I can help!

**Comment by securestep9** [comment]
@zekeyeager-tech same question to you as to other students keen to get this assigned to them -  Do you understand the current architecture and places in the code which created this challenge?   Do you use Python on Windows? Do you have experience porting Python code from Linux to Windows ?

**Comment by codeshazard** [comment]
Hey, I see that there are some PRs open for this issue. Is there any additional work or testing needed that I can help with? I am a beginner and I would like to work on this good first issue!

**Comment by pUrGe12** [comment]
@codeshazard Hello! Could you test if PR #1137 works well on windows? You can make any path fixes along the way

**Comment by codeshazard** [comment]
@pUrGe12 Sure, will do! Thanks for the opportunity.

**Comment by codeshazard** [comment]
@pUrGe12 
Hi! I tested PR #1137 on Windows 11 as requested. Initially, it didn’t work out-of-the-box — I ran into issues with path handling, language detection, and CLI argument parsing. I made several changes to get it working, including:
- Refactoring path logic to support Windows-style separators
- Cleaning up module name formatting and language detection
- Patching the -v argument to accept verbosity levels
After these fixes, the scan completed successfully on Windows.
Would it be okay if I open a separate PR to share these changes? It would document the platform-specific adjustments and help improve cross-platform support.

**Comment by securestep9** [comment]
@codeshazard yes, please open a separate PR

**Comment by vishantgawali1811** [comment]
Hi! I’m new to open source and would like to work on this issue as my first contribution. Could you please assign it to me or guide me on how to start?

**Comment by securestep9** [comment]
Hi @vishantgawali1811 - this feature is already in active development by @codeshazard I suggest you look at other OWASP Projects in the need of contributions here: https://nest.owasp.org/contribute

**Comment by Awaneesh03** [comment]
Hey, I’d like to work on this

**Comment by koffandaff** [comment]
Hi @securestep9  I just saw this good first issue not answered yet and currently under work is there a way to help or assist to resolve this issue with @codeshazard  to resolve it with better efficiency 

I look forward to help @codeshazard  and resolve this issue

**Comment by BRGOVIND** [comment]
“Hi! I would like to work on this issue and refactor path logic using pathlib so it works on all OSes. Please let me know if there is any specific guidance.”

**Comment by Shashwat-Acharya** [comment]
Are there additional good first issues available for someone who wants to contribute to the project’s logic layer?
